R/test-threads.R

```diff
@@ -330,4 +297,123 @@ delete_models <- function(.mods, .tags = "test threads", .force = FALSE){
 }
 
 
+#' Parse model files and overwrite estimation iterations
+#'
+#' @param .mods a list of model objects. Generally created by `test_threads()`.
+#' @param .cap_iterations Max number of iterations for NONMEM to run.
+#'
+#' @importFrom readr parse_number
+#' @importFrom checkmate assert_int
+#' @importFrom stringr str_detect
+#' @importFrom rlang is_empty
+#'
+#' @keywords internal
+adjust_estimation_options <- function(.mods, .cap_iterations){
+
+  if(!is.null(.cap_iterations)){
+    assert_int(.cap_iterations, lower = 1)
+    .detect = c("MAXEVAL", "NITER", "NBURN", "MAX")
+    search_str <- paste(.detect, collapse = "|")
+    map(.mods, function(.mod){
+
+      mod_path <- get_model_path(.mod)
+      mod_lines <- mod_path %>% readLines() %>%
+        suppressSpecificWarning("incomplete final line found")
+
+
+      # Identify EST Blocks
+      est_idxs <- get_est_idx(mod_lines)
+
+      est_block <- map(seq_along(est_idxs), ~{
+        mod_lines[est_idxs[[.x]]]
+      })
+
+      est_idxs <- unlist(est_idxs)
+
+      if(is.null(est_idxs)){
+        warning(glue("No Estimation line found in {basename(mod_path)}, so there are no options to cap"))
+        return(NULL)
+      }
+
+      for(i in seq_along(est_block)){
+        est_block.i <- est_block[[i]]
+
+        # Check for cases where options cannot be declared for the same EST method
```

Comment:
I think we should delete this check because it doesn't matter to us (this function) whether there are both. We're really just checking for valid NONMEM code here which 1. isn't this function's job and 2. I'm not even that confident that we know what valid NONMEM code is in this circumstance.

Reply:
We can also delete the corresponding test and setup [here](https://github.com/metrumresearchgroup/bbr/blob/f6670d88c6a1e7fe73c4cbaa6dd78829eeb11ccb/tests/testthat/test-test-threads.R#L182-L220)